src/index.ts

```diff
@@ -192,6 +193,16 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
                         document.title = htmlFileConfiguration.title
                     }
 
+                    if (htmlFileConfiguration.favicon) {
+                        // Injects a favicon if present
+                        fs.copyFile( htmlFileConfiguration.favicon, `${outdir}/favicon.ico` )
```

Comment:
copyFile is async, so it needs to be awaited here. :)

```suggestion
                        await fs.copyFile( htmlFileConfiguration.favicon, `${outdir}/favicon.ico` )
```